### PR TITLE
CyberArkEPM: first commit

### PR DESCRIPTION
**Pull request by averbn**
Change(s):
   - Add new Dataconnector for CyberArk EPM.

### CODE REVIEW

File: Sample Data/Custom/CyberArkEPM_CL.json [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+       "applicationSubType":null,
+       "fileVersion":"22.33.213.2",
+       "modificationTime":"2022-03-09T06:15:26.532Z",
+       "userIsAdmin":false,
+       "agentEventCount":1,
+       "skippedCount":0,
+       "accessAction":"false",
+       "accessTargetType":"Internet",
+       "accessTargetName":null,
+       "processCommandLine":"-Embedding",
```
Comment (v-sabiraj) [inline Sample Data/Custom/CyberArkEPM_CL.json:69]: Hey @averbn , in a parser it is given as guid( _g) for "processCommandLine" but in Sample Data the value is string, please use GUID in sample data or use "_s" for processCommandLine in parser, please check and update, thanks.
image

File: Solutions/CyberArkEPM/Parsers/CyberArkEPM [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// Usage Instructions:
```
Comment (v-sabiraj) [inline Solutions/CyberArkEPM/Parsers/CyberArkEPM:1]: @averbn , Documentation links should not include locale: Solutions/CyberArkEPM/Parsers/CyberArkEPM, please check and update.
Add **.txt** extenstion to the parser file, thanks.

File: Solutions/CyberArkEPM/DataConnectors/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# DO NOT include azure-functions-worker in this file
+# The Python Worker is managed by Azure Functions platform
+# Manually managing azure-functions-worker may cause unexpected issues
+
+azure-storage-file-share==12.5.0
```
Comment (v-sabiraj) [inline Solutions/CyberArkEPM/DataConnectors/requirements.txt:5]: Hey @averbn, the version for "azure-storage-file-share" is 12.5.0 , but not updated in [CyberArkEPMSentinelConn.zip ](https://github.com/socprime/Azure-Sentinel/blob/CyberArkEPM_Dataconnector/Solutions/CyberArkEPM/DataConnectors/CyberArkEPMSentinelConn.zip?raw=true), it is "azure_storage_file_share-12.3.0.dist-info", please check and update , thanks.

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

File: Solutions/CyberArkEPM/DataConnectors/CyberArkEPM_API_FunctionApp.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+						],
+						"label": "Primary Key"
+					},
+					"type": "CopyableLabel"
+				}
+			]
+		},
+		{
+			"title": "Option 1 - Azure Resource Manager (ARM) Template",
+			"description": "Use this method for automated deployment of the CyberArk EPM data connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinel-CyberArkEPMAPI-azuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n> **NOTE:** Within the same resource group, you can't mix Windows and Linux apps in the same region. Select existing resource group without Windows apps in it or create new resource group.\n3. Enter the **CyberArkEPMUsername**, **CyberArkEPMPassword**, **CyberArkEPMServerURL**  and deploy. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
```
Comment (v-sabiraj) [inline Solutions/CyberArkEPM/DataConnectors/CyberArkEPM_API_FunctionApp.json:106]: Please fix the typos "Tempate" to "Template", thanks.

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

**Comment by v-sabiraj** [comment]
Hey @oshezaf , will you please review parser, thanks.

File: Sample Data/Custom/CyberArkEPM_CL.json (hunk 1 above)
Comment (averbn) [reply Sample Data/Custom/CyberArkEPM_CL.json:69]: I don`t understand what I should change: we have two fields processCommandLine_g and processCommandLine_s. One for guide and second for process cmd. I think both should be with string Type. I see that the field processCommandLine_g is missing in the examples - I will add it

File: Solutions/CyberArkEPM/DataConnectors/CyberArkEPM_API_FunctionApp.json (hunk 4 above)
Comment (averbn) [reply Solutions/CyberArkEPM/DataConnectors/CyberArkEPM_API_FunctionApp.json:106]: Done

File: Solutions/CyberArkEPM/Parsers/CyberArkEPM (hunk 2 above)
Comment (averbn) [reply Solutions/CyberArkEPM/Parsers/CyberArkEPM:1]: Done

File: Sample Data/Custom/CyberArkEPM_CL.json [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+       "policyName":null,
+       "hash":"420BE0941CEB6B0781A93D6ED81B1C80E733A5FF",
+       "publisher":"Microsoft Corporation",
+       "eventType":"Launch",
+       "sourceType":"Updater",
+       "sourceName":"Updater (Windows Update)",
+       "lastEventDate":"2022-03-10T13:32:03.206Z",
+       "firstEventDate":"2022-03-10T13:32:03.206Z",
+       "userName":"DESKTOP-S4J3FEM\\Adam",
+       "justification":null,
```
[the remark this answers is not in the document]
Comment (averbn) [reply Sample Data/Custom/CyberArkEPM_CL.json:46]: Done

**Comment by v-sabiraj** [comment]
Hey @averbn , the checks are failing "JsonFileValidation" and "sampleDataValidator", this may be because of some typos for """e":"-Embedding", "source""e":"", will you please check and update, thanks.
image

**Comment by v-sabiraj** [comment]
Hey @averbn , Tested the data connector by deploying the functions. Can see unexpected error for "CyberArkEPMSentinelConnector" function. Please check and update. Thanks.
image

**Comment by v-sabiraj** [comment]
Hey @averbn , will you please resolve the requested changes, thanks.

**Comment by averbn** [comment]
> Hey @averbn , will you please resolve the requested changes, thanks.

It seems I have no permission for doing that.

**Comment by v-sabiraj** [comment]
Hey @averbn, you need to change "processCommandLine":"f67fa0d1-30a-45c0-b199-1efd2" to correct GUID, the Guid has 32 digits. 
image
as it was before ,in sample data, please check and update, thanks.

File: Solutions/CyberArkEPM/Parsers/CyberArkEPM.txt [hunk 6]
```diff
@@ -0,0 +1,92 @@
+// Usage Instructions: 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name as CyberArkEPM and specifying Legacy Category.
+// This function maps CyberArk EPM events to Azure Sentinel Information Model (ASIM) (https://docs.microsoft.com/en-us/azure/sentinel/normalization).
```
Comment (v-sabiraj) [inline Solutions/CyberArkEPM/Parsers/CyberArkEPM.txt:3]: Documentation links should not include locale: (https://docs.microsoft.com/en-us/azure/sentinel/normalization), please check and update.

File: Solutions/CyberArkEPM/Parsers/CyberArkEPM.txt [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                 LogonAttemptTypeId=logonAttemptTypeId_d,
+                 LogonStatusId=logonStatusId_d,
+                 SrcFileMimeType=mimeType_s,
+                 ModificationTime=modificationTime_t,
+                 ActingProcessFileOriginalName=originalFileName_s,
+                 Owner=owner_s,
+                 PackageName=packageName_s,
+                 PolicyId=policyId_d,
+                 PolicyName=policyName_s,
+                 ActingProcessGuid=processCommandLine_g,
```
Comment (v-sabiraj) [inline Solutions/CyberArkEPM/Parsers/CyberArkEPM.txt:59]: Hey @averbn, in sample data there is not any guid used for "processCommandLine", will you please check and remove this, thanks

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

File: Solutions/CyberArkEPM/DataConnectors/azuredeploy_Connector_CyberArkEPM_API_AzureFunction.json [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+                        "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=', toLower(variables('FunctionName')),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', toLower(variables('FunctionName'))), '2019-06-01').keys[0].value, ';EndpointSuffix=',toLower(variables('StorageSuffix')))]",
+                        "logAnalyticsUri": "[variables('LogAnaltyicsUri')]",
+                        "WorkspaceID": "[parameters('WorkspaceID')]",
+                        "WorkspaceKey": "[parameters('WorkspaceKey')]",
+                        "CyberArkEPMUsername": "[parameters('CyberArkEPMUsername')]",
+                        "CyberArkEPMPassword": "[parameters('CyberArkEPMPassword')]",
+                        "CyberArkEPMServerURL": "[parameters('CyberArkEPMServerURL')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/averbn/azure_sentinel_data_connectors/raw/main/cyberarkepm-azure-sentinel-data-connector/CyberArkEPMSentinelConn.zip?raw=true"
```
Comment (v-sabiraj) [inline Solutions/CyberArkEPM/DataConnectors/azuredeploy_Connector_CyberArkEPM_API_AzureFunction.json:162]: Hey @averbn, just change "WEBSITE_RUN_FROM_PACKAGE" link to https://aka.ms/sentinel-CyberArkEPMAPI-functionapp, thanks.

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

**Comment by v-sabiraj** [comment]
Hey @averbn, everything else looks good, just change "WEBSITE_RUN_FROM_PACKAGE" link to https://aka.ms/sentinel-CyberArkEPMAPI-functionapp, thanks.

File: Solutions/CyberArkEPM/DataConnectors/azuredeploy_Connector_CyberArkEPM_API_AzureFunction.json (hunk 8 above)
Comment (averbn) [reply Solutions/CyberArkEPM/DataConnectors/azuredeploy_Connector_CyberArkEPM_API_AzureFunction.json]: Done, sorry for delay

**Pull request merged**